app/src/main/java/dk/mifu/pmos/vegetablegardening/fragments/dialogs/CreatePlantDialogFragment.kt

```diff
@@ -0,0 +1,203 @@
+package dk.mifu.pmos.vegetablegardening.fragments.dialogs
+
+import android.app.DatePickerDialog
+import android.os.Bundle
+import android.view.*
+import android.widget.ArrayAdapter
+import android.widget.Toast
+import androidx.appcompat.app.AppCompatActivity
+import androidx.fragment.app.DialogFragment
+import androidx.navigation.NavOptions
+import androidx.navigation.fragment.findNavController
+import androidx.navigation.fragment.navArgs
+import dk.mifu.pmos.vegetablegardening.R
+import dk.mifu.pmos.vegetablegardening.database.AppDatabase
+import dk.mifu.pmos.vegetablegardening.database.GardenRepository
+import dk.mifu.pmos.vegetablegardening.database.PlantRepository
+import dk.mifu.pmos.vegetablegardening.databinding.FragmentCreatePlantDialogBinding
+import dk.mifu.pmos.vegetablegardening.models.Bed
+import dk.mifu.pmos.vegetablegardening.models.Plant
+import kotlinx.coroutines.*
+import java.text.SimpleDateFormat
+import java.util.*
+
+class CreatePlantDialogFragment : DialogFragment() {
+    private lateinit var binding: FragmentCreatePlantDialogBinding
+
+    private val args: CreatePlantDialogFragmentArgs by navArgs()
+
+    private var plant: Plant = Plant("")
+
+    override fun onCreateView(inflater: LayoutInflater, container: ViewGroup?, savedInstanceState: Bundle?): View {
+        binding = FragmentCreatePlantDialogBinding.inflate(inflater, container, false)
+
+        setDataFromArgs()
+        setDatePickerListeners()
+        addOptionsToPlantSowing()
+        setSaveButtonListener()
+        setCancelButtonListener()
+
+        return binding.root
+    }
+
+    private fun setDataFromArgs() {
+        val argPlant = args.plant
```

Comment:
I don't understand why the fragment for creating new plants takes a plant as parameter?

Reply:
Rename to CustomPlantDialogFragment